tasks/ast_tools/src/generators/ast_kind.rs

```diff
@@ -87,6 +87,7 @@ impl Generator for AstKindGenerator {
         let mut span_match_arms = quote!();
         let mut address_match_arms = quote!();
         let mut as_methods = quote!();
+        let mut ast_type_max_usize: usize = 0;
```

Comment:
I think you could just use `next_index` here. If the problem is printing it without the `u16` postfix, you can use this utility function:

https://github.com/oxc-project/oxc/blob/242e7210aa37e6eef14457b28aebbc31035d3c28/tasks/ast_tools/src/utils.rs#L74-L81

```rs
let ast_type_max = number_lit(next_id);
quote! {
    pub const AST_TYPE_MAX: usize = #ast_type_max;
}
```

Not that it matters much in this case. But this auto-postfix behavior of `quote!` drove me mad for ages until I figured out this workaround, so just letting you know too...

Reply:
I'd recommend sticking with `usize` for the const. You'll be comparing it to other `usizes`.

Also, I made a mistake. It should be `let ast_type_max = number_lit(next_id - 1);`.